packages/codelabs/src/basics/lit-html.md

```diff
@@ -0,0 +1,1139 @@
+
+<detail>
+  <summary>View final result</summary>
+
+```html
+<!DOCTYPE html>
+<html>
+  <head></head>
+
+  <body>
+    <todo-app></todo-app>
+
+    <script type="module">
+      ...code sample goes here...
+    </script>
+  </body>
+</html>
+```
+
+</detail>
+
+## Create a child component
+
+It looks like we're feature complete! We can display a list of todos, add or remove them and check them off as we finish them.
+
+However we've been putting everything into one component, and it's starting to get crowded. It's better to split things up into separate components, we're building web components after all.
+
+The todo list would be an ideal candidate for moving into a separate component. Let's go ahead and define the basic structure of our element. If you're working in an online editor, it's probably easiest to do this in the same file. If you're using a local editor, it's best to create a separate file for this.
+
+If you use a separate file, don't forget to import `LitElement` and `html` for that module as well.
+
+```js
+class TodoList extends LitElement {
+  static get properties() {
+    return {
+      todos: { type: Array },
+    };
+  }
+
+  render() {
+    if (!this.todos) {
+      return html``;
+    }
+
+    return html`
+      <ol>
+        ${this.todos.map(
+          todo => html`
+            <li>
+              <input
+                type="checkbox"
+                .checked=${todo.finished}
+                @change=${e => this._changeTodoFinished(e, todo)}
+              />
+              ${todo.text}
+              <button @click=${() => this._removeTodo(todo)}>X</button>
+            </li>
+          `,
+        )}
+      </ol>
+    `;
+  }
+}
+
+customElements.define('todo-list', TodoList);
+```
+
+Ok so the structure of the class should be pretty familiar now, there is a `todos` property and a template to render. The template looks almost the same as the previous, except that now there is an if statement at the start. We need this because unlike before, our todo list isn't the one in charge of the data.
+
+The parent element is still in charge, and we expect it to pass along the todos list to this component. This means that we cannot assume the list to always be there when we do a render. If we don't take care of this somehow, our component will crash because you cannot run a `map` function on `undefined.` Adding early returns in your render function is a simple way to do this, it's easy to see which properties are required for rendering.
+
+Next, we need to somehow let the parent element know that the user clicked on the checkbox or remove the button. We can do this using DOM events. DOM events are great because the structure and hierarchy of our application are reflected in the DOM, so when a component fires an event only it's parent components can receive them. This creates an automatic scoped system.
```

Comment:
```suggestion
Next, we need to somehow let the parent element know that the user clicked on the checkbox or remove the button. We can do this using DOM events. DOM events are great because the structure and hierarchy of our application are reflected in the DOM, so when a component fires an event only its parent components can receive them. This creates an automatic scoped system.
```

`it's` stands for `it is`
`its` is possessive